### ISSUE TITLE
Error when running amq.py

### ISSUE THREAD

**Issue by Korgi2** [stephen-huan/anime-music-quiz#6]
Traceback (most recent call last):
  File "A:\anime-music-quiz-master\anime-music-quiz-master\amq.py", line 11, in <module>
    import main
  File "A:\anime-music-quiz-master\anime-music-quiz-master\main.py", line 4, in <module>
    import db
  File "A:\anime-music-quiz-master\anime-music-quiz-master\db.py", line 108, in <module>
    db = load_db()
  File "A:\anime-music-quiz-master\anime-music-quiz-master\db.py", line 29, in load_db
    return json.load(f)
  File "C:\Program Files\WindowsApps\PythonSoftwareFoundation.Python.3.9_3.9.2288.0_x64__qbz5n2kfra8p0\lib\json\__init__.py", line 293, in load
    return loads(fp.read(),
  File "C:\Program Files\WindowsApps\PythonSoftwareFoundation.Python.3.9_3.9.2288.0_x64__qbz5n2kfra8p0\lib\json\__init__.py", line 346, in loads
    return _default_decoder.decode(s)
  File "C:\Program Files\WindowsApps\PythonSoftwareFoundation.Python.3.9_3.9.2288.0_x64__qbz5n2kfra8p0\lib\json\decoder.py", line 337, in decode
    obj, end = self.raw_decode(s, idx=_w(s, 0).end())
  File "C:\Program Files\WindowsApps\PythonSoftwareFoundation.Python.3.9_3.9.2288.0_x64__qbz5n2kfra8p0\lib\json\decoder.py", line 355, in raw_decode
    raise JSONDecodeError("Expecting value", s, err.value) from None
json.decoder.JSONDecodeError: Expecting value: line 1 column 1 (char 0)

Process finished with exit code 1

**Comment by stephen-huan** [comment]
The error seems to be `db.py` can't parse `info.json`. What's the content of `info.json`?
If it's empty, try putting `{}` into `info.json`.

**Comment by Korgi2** [comment]
yeah that fixed it but now im stuck on it not progressing beyond the login question unfortunately. I'd run it outside of pycharm but my pc really hates python.
![image](https://user-images.githubusercontent.com/61196219/141665749-8ed200c7-cdd4-4c66-a690-ce8e8391471f.png)

**Comment by stephen-huan** [comment]
hmm, that's odd. If it's asking for a username, the very next step is to call `getpass.getpass()`. Might be an issue with [getpass](https://docs.python.org/3/library/getpass.html) on Windows if you're not in a terminal? Try replacing the `getpass.getpass()` call on line 26 in [`amq.py`](https://github.com/stephen-huan/anime-music-quiz/blob/fac298bfb347d8236edafc058849b1e7be18a5ca/amq.py#L26) with something like:
```python
# prompt for information
else:
    USER, PASS = input("Username: "), input("Password: ")
```
or specify a `login.json` file as follows:
```json
{
    "username": "USERNAME",
    "password": "PASSWORD"
}
```

**Comment by Korgi2** [comment]
![image](https://user-images.githubusercontent.com/61196219/141699447-4e02fda0-89e8-4bcc-8371-48bfb4835f64.png)
it seems to be missing a prompt.

**Comment by Korgi2** [comment]
Traceback (most recent call last):
  File "C:\Users\Korgi\AppData\Local\Packages\PythonSoftwareFoundation.Python.3.9_qbz5n2kfra8p0\LocalCache\local-packages\Python39\site-packages\selenium\webdriver\common\service.py", line 74, in start
    self.process = subprocess.Popen(cmd, env=self.env,
  File "C:\Program Files\WindowsApps\PythonSoftwareFoundation.Python.3.9_3.9.2288.0_x64__qbz5n2kfra8p0\lib\subprocess.py", line 951, in __init__
    self._execute_child(args, executable, preexec_fn, close_fds,
  File "C:\Program Files\WindowsApps\PythonSoftwareFoundation.Python.3.9_3.9.2288.0_x64__qbz5n2kfra8p0\lib\subprocess.py", line 1420, in _execute_child
    hp, ht, pid, tid = _winapi.CreateProcess(executable, args,
FileNotFoundError: [WinError 2] The system cannot find the file specified

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "A:\anime-music-quiz-master\anime-music-quiz-master\amq.py", line 118, in <module>
    driver = login()
  File "A:\anime-music-quiz-master\anime-music-quiz-master\amq.py", line 39, in login
    driver = webdriver.Chrome(options=chrome_options)
  File "C:\Users\Korgi\AppData\Local\Packages\PythonSoftwareFoundation.Python.3.9_qbz5n2kfra8p0\LocalCache\local-packages\Python39\site-packages\selenium\webdriver\chrome\webdriver.py", line 69, in __init__
    super(WebDriver, self).__init__(DesiredCapabilities.CHROME['browserName'], "goog",
  File "C:\Users\Korgi\AppData\Local\Packages\PythonSoftwareFoundation.Python.3.9_qbz5n2kfra8p0\LocalCache\local-packages\Python39\site-packages\selenium\webdriver\chromium\webdriver.py", line 90, in __init__
    self.service.start()
  File "C:\Users\Korgi\AppData\Local\Packages\PythonSoftwareFoundation.Python.3.9_qbz5n2kfra8p0\LocalCache\local-packages\Python39\site-packages\selenium\webdriver\common\service.py", line 84, in start
    raise WebDriverException(
selenium.common.exceptions.WebDriverException: Message: 'chromedriver' executable needs to be in PATH. Please see https://chromedriver.chromium.org/home


Process finished with exit code 1


I apologize as I have 0 idea about any code and that includes Python. I just know this is the error I get after fixing the no password prompt.

**Comment by stephen-huan** [comment]
Oops, forgot to put the webdriver dependancy in the README.

You need a browser driver for [Selenium](https://www.selenium.dev/documentation/getting_started/installing_browser_drivers/), in this case I was using [chromedriver](https://sites.google.com/chromium.org/driver/); here's the [downloads](https://sites.google.com/chromium.org/driver/downloads) page. Once you get a `chromedriver` binary you'll have to add it to your `PATH` so that Selenium can find it; instructions are provided on the Selenium page under in the section "Adding Executables to your PATH".

Basically Selenium is library that lets code control a browser like a user would, but it needs a browser driver to interact with the browser. You can also use something like [geckodriver](https://github.com/mozilla/geckodriver) if you replace `webdriver.Chrome()`in [amq.py](https://github.com/stephen-huan/anime-music-quiz/blob/fac298bfb347d8236edafc058849b1e7be18a5ca/amq.py#L39) with `webdriver.Firefox()`.

**Comment by Korgi2** [comment]
At risk of being stupid, where in my Python 39 folder do I put the exe?

**Comment by stephen-huan** [comment]
The executable can be wherever you want, I think Selenium suggests `C:\WebDriver\bin`. The important part is that the `PATH` environmental variable is set to the correct folder containing the executable.

Again, instructions on modifying `PATH` can be found on the Selenium [documentation](https://www.selenium.dev/documentation/getting_started/installing_browser_drivers/).

**Comment by Korgi2** [comment]
I did what the website said to change the path to webdriver/bin but i got the same error when i run the py
![image](https://user-images.githubusercontent.com/61196219/141705134-1460bb39-50d3-4138-bae0-25d5fefe7209.png)
![image](https://user-images.githubusercontent.com/61196219/141705144-a3e755a1-658f-4906-bb32-71db406a5771.png)

Whats weird is when i test in cmd it works
![image](https://user-images.githubusercontent.com/61196219/141705182-2b736191-a475-42f7-8902-9db1c6ed9ada.png)
but in pycharm it gives an error

**Comment by stephen-huan** [comment]
Interesting. Can you try to run `python` outside of PyCharm? I suspect what makes `python` hard to run is also the reason why adding `geckodriver` to `PATH` doesn't register with `python`.

**Comment by Korgi2** [comment]
The main reason im using pycharm is ive been unable to launch .py files outside of it. The python terminal pops up for a brief moment and shuts itself for no clear reason. The same issue occurs when I attempt to run files within powershell.

**Comment by Korgi2** [comment]
I figured out why it wouldn't run but now running it in cmd I have a different issue.

![image](https://user-images.githubusercontent.com/61196219/141865561-72ffc26f-f10c-498e-ba13-e67918daa55b.png)

**Comment by stephen-huan** [comment]
Just remove the options, you don't need them for the bot to work. Initialize the webdriver with just
```python
driver = webdriver.Firefox()
```

**Comment by stephen-huan** [comment]
Just remove the options, you don't need them for the bot to work. Initialize the webdriver with just
```python
driver = webdriver.Firefox()
```

**Comment by Korgi2** [comment]
I'm sorry to continously have issues but I think we're almost through with it. The app opens like 5 firefox windows before loading amq then crashing

![image](https://user-images.githubusercontent.com/61196219/141874620-08b437ea-15d3-4322-9abd-4187fb211ffb.png)


I am really unsure why I am having so many issues in a row like this. The program seemed to work perfectly in your demonstration video, idk why my PC is like this. Do you have a discord? Might be faster.

**Comment by stephen-huan** [comment]
Feel free to email if you want to get in touch, I'll send my Discord.
